### PR TITLE
added api documentation for layers

### PR DESCRIPTION
**Pull request by devcrocod**
Added API documentation for all layers, except statistical

### CODE REVIEW

**Comment by AndreiKingsley** [comment]
In some places it is necessary to specify what the attribute is responsible for (within a geom). For example `color`/`fill` in points changes its behaviour depending on the symbol; `x`/`y` in raster defines rectangle centres; and so on.

**Comment by AndreiKingsley** [comment]
You can base this on the attributes documentation in lets-plot and ggplot2

**Comment by AndreiKingsley** [comment]
In general, the attribute description is not connected in any way to the geom in which section it is described, and this is strongly noticeable. 
I think for the initial version it's still ok, but if the attribute does something specific, it's worth specifying it additionally

File: docs/topics/apiRef/properties.topic [hunk 1]
```diff
@@ -0,0 +1,828 @@
...
+        <p>
+            The <code>symbol</code> aesthetic is integral to plotting, especially in visualizations that rely on
+            markers, such as scatter plots and point ranges. This aesthetic enables the customization of marker shapes,
+            allowing for a diverse representation of data points.
+        </p>
+
+        <p><b>Setting</b></p>
+        <list>
+            <li>
+                <p><code>symbol = Symbol</code>: Assigns a specific marker shape to all points within a layer:</p>
```
Comment (AndreiKingsley) [inline docs/topics/apiRef/properties.topic:366]: Images here?

File: docs/topics/apiRef/properties.topic [hunk 2]
```diff
@@ -0,0 +1,828 @@
...
+            The type aesthetic is used to specify the style of lines in a plot, allowing for a diverse range of visual
+            representations.
+            This aesthetic enables the customization of line styles, providing clarity and distinction in data
+            visualizations.
+        </p>
+
+        <p><b>Setting</b></p>
+        <list>
+            <li>
+                <p><code>type = LineType</code>: applies a specific line style to lines within a layer.
```
Comment (AndreiKingsley) [inline docs/topics/apiRef/properties.topic:433]: Images

**Comment by AndreiKingsley** [comment]
Should we add images to this section?
Should we place images of possible point symbols and line type here or in some other section?

File: docs/topics/apiRef/properties.topic (hunk 1 above)
Comment (devcrocod) [reply docs/topics/apiRef/properties.topic:366]: Where is better to get images from?

**Comment by devcrocod** [comment]
I think yes, images would help a lot. But I don’t know where it’s best to get them so that they suit us. Only if I draw it myself

File: docs/topics/apiRef/properties.topic (hunk 1 above)
Comment (AndreiKingsley) [reply docs/topics/apiRef/properties.topic:366]: image

Comment (AndreiKingsley) [reply docs/topics/apiRef/properties.topic:366]: this is default image for ggplot2

File: docs/topics/apiRef/properties.topic (hunk 2 above)
Comment (AndreiKingsley) [reply docs/topics/apiRef/properties.topic:433]: image

**Comment by AndreiKingsley** [comment]
So far I don't understand the best solution for images either, maybe it's worth discussing.

File: docs/topics/apiRef/properties.topic (hunk 1 above)
Comment (devcrocod) [reply docs/topics/apiRef/properties.topic:366]: we can't use them, at least they require preprocessing: remove numbers, change colors

File: docs/topics/apiRef/properties.topic (hunk 2 above)
Comment (devcrocod) [reply docs/topics/apiRef/properties.topic:433]: Unfortunately, these images cannot be inserted within writerside

**Comment by devcrocod** [comment]
I created issue #331 on images because it requires additional work in graphics editors such as figma

**Comment by AndreiKingsley** [comment]
Also, "line type" should be written in two words everywhere. In API names `LineType` and `lineType` are used since it's written in 2 words.

**Comment by devcrocod** [comment]
> In some places it is necessary to specify what the attribute is responsible for (within a geom). For example color/fill in points changes its behaviour depending on the symbol; x/y in raster defines rectangle centres; and so on.

As you can see, I reused descriptions for many layers, so I wrote the most general definitions. Otherwise it would take even longer.

> You can base this on the attributes documentation in lets-plot and ggplot2

I looked at the ggplot2 documentation and based partly on it

>In general, the attribute description is not connected in any way to the geom in which section it is described, and this is strongly noticeable.
I think for the initial version it's still ok, but if the attribute does something specific, it's worth specifying it addition


You are right, they are not tied to geoms. Most of the definitions are in `properties.topic` and pasted from there. I think there are no problems with describing aes. This problem should be solved by examples for each aes, this issue #315 about this

**Comment by devcrocod** [comment]
> Also, "line type" should be written in two words everywhere. In API names LineType and lineType are used since it's written in 2 words.

Oh, I just applied suggestions from grazie

**Review by AndreiKingsley — APPROVED** [review]

**Pull request merged**